webapp/src/index.js

```diff
@@ -19,6 +19,10 @@ export default class Plugin {
             (channelId) => {
                 store.dispatch(openMeetingSettingsModal(channelId));
             });
+
+        registry.registerPostDropdownMenuAction('Re-queue', (params) => {
+            store.dispatch(requeueItem(params));
+        });
```

Comment:
We should only show this if the post has an agenda item right? We can use the third argument to [registerPostDropdownMenuAction](https://developers.mattermost.com/integrate/plugins/webapp/reference/#registerPostDropdownMenuAction) here.

Is it feasible to read the post's message and determine that way? We may not have the channel's agenda settings readily available. We may need to start putting something in the post's props to signal that this post is an agenda item if this isn't possible.

Reply:
we could make channel's agenda settings readily available but 2/5 in my opinion we should go with your other suggestion i.e props in message. It would solve multiple issues like in current implementation we can't locate agenda messages if user change hashtag format multiple times. Also, backwards compatibility could be added after this.
I am not sure if that's possible, would be great if we could store something like this in props:
`{
isAgendaItem:true,
hastagDateFormat: "dd/mm/yy",
hashtagPrefix:"myprefix",
eventDate:"isoDate"
}`
@mickmister thoughts?

Reply:
@sanjaydemansol Yeah in the props we can do something like:

```json
{
    "agendaSettings": {
        "hasftagDateFormat": "dd/mm/yy",
        "hashtagPrefix": "myprefix",
        "eventDate": "isoDate"
    }
}
```

Then we just need to check `post.props.agendaSettings`. If that's not in the props, don't show the requeue button.

What is the `eventDate` you're thinking of? Is that something we'll need to keep up to date in the post's props? When will we reference the props instead of the channel's agenda settings and vise versa?

Reply:
It will be the date the even is supposed to happen on, since hashtag format may be changed any time. Also user may change format after scheduling a meeting rendering old items unparsable. Ex, #1_2_2021 could mean any of two dates. So, either we must have old hashtag format(which was used at the time of creation of this hashtag) or better `eventDate`.